src/app/services/dataset/dataset_processor_service.py

```diff
@@ -0,0 +1,158 @@
+import uuid
+
+from config.config_factory import ConfigFactory
+from logging_config import logging
+from models.dataset_models import (
+    DatasetMetadataWithoutId,
+    NewDatasetMetadata,
+    NewDatasetWithMetadata,
+    UnitDataset,
+)
+from repositories.dataset_repository import DatasetRepository
+from services.dataset.dataset_writer_service import DatasetWriterService
+from services.datetime_service import DatetimeService
+
+config = ConfigFactory.get_config()
+
+
+logger = logging.getLogger(__name__)
+
+
+class DatasetProcessorService:
+    def __init__(self) -> None:
+        self.dataset_repository = DatasetRepository()
+
+        self.dataset_writer_service = DatasetWriterService(self.dataset_repository)
+        pass
+
+    def process_new_dataset(
+        self, filename: str, new_dataset: NewDatasetWithMetadata
+    ) -> None:
+        """
+        Processes the incoming dataset.
+
+        Parameters:
+        filename (str): the filename of the json containing the dataset data
+        new_dataset (NewDatasetWithMetadata): new dataset to be processed
+        """
+        logger.info("Processing new dataset...")
+        logger.debug(f"Dataset being processed: {new_dataset}")
```

Comment:
Im not sure how i feel about logging out the datasets here and other places in this function. Even if there isnt security/GDPR reasons not too i just feel like it clutters the output. Could be worth changing to the id to allow us to still identify whats being processed but keep it a bit more compact

Reply:
Changed it to log dataset id instead